lib/client/conntest/connection_tester.go

```diff
@@ -29,12 +31,24 @@ import (
 // - additional paramenters which depend on the actual kind of resource to test
 // As an example, for SSH Node it also includes the User/Principal that will be used to login.
 type TestConnectionRequest struct {
+	// ResourceKind describes the type of resource to test.
 	ResourceKind string `json:"resource_kind"`
+	// ResourceName is the identification of the resource's instance to test.
 	ResourceName string `json:"resource_name"`
 
+	// ProxyHostPort is the proxy to use in the `--proxy` format (host:webPort,sshPort)
+	ProxyHostPort string `json:"-"`
+
+	// TLSRoutingEnabled indicates that proxy supports ALPN SNI server where
+	// all proxy services are exposed on a single TLS listener (Proxy Web Listener).
+	TLSRoutingEnabled bool `json:"-"`
```

Comment:
I'm not sure these really belong in the test connection request tbh. I'd probably move them to the config that connection tester is initialized with.